src/handlers.jl

```diff
@@ -213,15 +235,84 @@ function get_token(code, pos)
     end
     startpos = startpos < pos ? nextind(code, startpos) : pos
     if !is_id_start_char(code[startpos])
-        return ""
+        return startpos, -1, crossed_parentheses, stop
     end
     while endpos < lastindex(code) && is_id_char(code[endpos])
         endpos = nextind(code, endpos)
     end
     if !is_id_char(code[endpos])
         endpos = prevind(code, endpos)
     end
-    return code[startpos:endpos]
+    return startpos, endpos, crossed_parentheses, stop
+end
+
+function get_token(code, pos)
+    """
+        get_token(code, pos)
+
+    Given a string and a cursor position, find substring to request
+    help on by:
+
+    1. Searching backwards for the closest token (may be invalid)
+    2. Keep searching backwards until we find an token before an unbalanced '('
+        a. If (1) is not valid, store the first valid token
+        b. We assume a token before an unbalanced '(' is a function
+    3. If we find a possible function token, return this token.
+    4. Otherwise, return the last valid token
+
+    # Important Note
+
+    Tokens are chosen following several empirical observations instead of rigorous rules.
+    We assume that the first valid token before left-imbalanced (more '(' than ')') parentheses is the function "closest" to cursor.
+    The following examples use '|' to denote cursor, showing observations on parentheses.
+
+    - `f()|` has balanced parentheses with nothing within, thus `f` is the desired token.
+    - `f(|)` has imbalanced parentheses, thus `f` is the desired token.
+    - `f(x|, y)` gives tokens `x` and `f`. `x` has balanced parentheses, while `f` is left-imbalanced. `f` is desired.
+    - `f(x)|` returns `f`
+    - `f(x, y)|` returns `f`.
+    - `f((x|))` returns `f`, as expected
+    - `f(x, (|y))` returns `f`. **This is a hack**, as I deduct `crossed_parentheses` whenever a separator is encountered, clamped to 0!
+        Otherwise, `x` would be returned.
+    - `f(x, (y|))`, `f(x, (y)|)`, and `f(x, (y))|` all behave as above. Arbitrary nesting of tuples should not cause misbehavior.
```

Comment:
What do we do if the argument is a more complicated expression, e.g. `f("foo bar")` or `f(3 + 5 -2)`?   Or `Vector{Int}(undef, n)`?

Reply:
`f("foo bar")` always returns `f` no matter where the cursor is (tested interactively).

`f(3 + 5 - 2)` always returns `f` no matter where the cursor is (also tested interactively).

`Vector{Int}(undef, n)` _happens to_ return `Vector` no matter where the cursor is. I did not account for `{}` in code, but it just happens to work.